operator/controllers/btpoperator_controller.go

```diff
@@ -95,25 +103,236 @@ var (
 	labelFilter = client.MatchingLabels{labelKeyForChart: operatorName}
 )
 
-type btpOperatorGvk struct {
-	APIVersion string `yaml:"apiVersion"`
-	Kind       string `yaml:"kind"`
-}
-
 // BtpOperatorReconciler reconciles a BtpOperator object
 type BtpOperatorReconciler struct {
 	client.Client
 	*rest.Config
 	Scheme                *runtime.Scheme
+	currentVersion        string
+	wereUpdateCheckDone   bool
 	WaitForChartReadiness bool
 }
 
+func (r *BtpOperatorReconciler) handleUpdate(ctx context.Context, cr *v1alpha1.BtpOperator) error {
+	logger := log.FromContext(ctx)
+
+	configMap := r.buildBtpManagerConfigMap()
+	err := r.Get(ctx, client.ObjectKeyFromObject(configMap), configMap)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return err
+	} else if err != nil && k8serrors.IsNotFound(err) {
+		configMap = nil
+	}
+
+	versionChanged, err := r.storeChartDetails(ctx, configMap)
+	if err != nil {
+		return err
+	}
+
+	logger.Info(fmt.Sprintf("version changed? -> %t", versionChanged))
+
+	if versionChanged {
+		ok, err := r.HandleReadyState(ctx, cr)
```

Comment:
inconsistency with CR state, CR is in `Processing` state while performing update